### DISCUSSION TITLE
How can I use code to insert records into a custom form table

### DISCUSSION THREAD

**Discussion by trexmo** [barseghyanartur/django-fobi#295] [Q&A] [answered]
I have a need to insert records into the table for a custom fobi form.  I tried using a requests.put() and that came close, I was able to put data but only if I grabbed the csrf tokens from the terminal and inserted them in the header in the put to my custom form using the DRF.
Is is possible to create a script that will insert the values without DRF and requests.put? I see how to do it with standard django models and views, but not custom forms.

Or is it better to modify the CsrfViewMiddleware to exempt csrf checking on my requests.put() to the custom fobi form?

**Comment by barseghyanartur** [comment, marked as answer]
I think in this case REST API is indeed your best option. It's strange though that you have to use CRF tokens. Django REST Framework doesn't require that normally.

**Reply by trexmo** [reply]
true, but only if the form is served by DRF as the csrf hidden field is included.  if asking from another application as in integration, that is when csrf kicks in.  i solved it by exempting specific paths from the csrf middleware.
